eclair-core/src/main/scala/fr/acinq/eclair/UInt64.scala

```diff
@@ -16,40 +16,42 @@
 
 package fr.acinq.eclair
 
-import java.math.BigInteger
-
+import com.google.common.primitives.UnsignedLongs
 import scodec.bits.ByteVector
+import scodec.bits._
 
-case class UInt64(private val underlying: BigInt) extends Ordered[UInt64] {
+case class UInt64(private val underlying: Long) extends Ordered[UInt64] {
 
-  require(underlying >= 0, s"uint64 must be positive (actual=$underlying)")
-  require(underlying <= UInt64.MaxValueBigInt, s"uint64 must be < 2^64 -1 (actual=$underlying)")
+  override def compare(o: UInt64): Int = UnsignedLongs.compare(underlying, o.underlying)
```

Comment:
I would move the `compareUnsigned` that's currently in `MilliSatoshi` and put it here: this way the use of `UnsignedLongs` is constrained to this file, which is really the only place where we want to hide it.
What do you think?

Reply:
I like this proposal, will do a `UInt64.compareUnsigned(other: MilliSatoshi)`

Reply:
bfd7f6b moves the unsigned comparison into the `UInt64` class, also the comparison now takes into account the possible negative values for MilliSatoshi and guarantees that `UInt64(_) > MilliSatoshi(y)` for any `y` smaller than `0`